modules/dxp/apps/commerce/commerce-machine-learning-impl/src/main/java/com/liferay/commerce/machine/learning/internal/recommendation/data/source/FrequentPatternCommerceMLRecommendationCPDataSourceImpl.java

```diff
@@ -67,10 +67,13 @@ public CPDataSourceResult getResult(
 			HttpServletRequest httpServletRequest, int start, int end)
 		throws Exception {
 
-		CommerceAccount commerceAccount =
-			commerceAccountHelper.getCurrentCommerceAccount(httpServletRequest);
+		long groupId = portal.getScopeGroupId(httpServletRequest);
 
-		if (commerceAccount == null) {
+		AccountEntry accountEntry =
+			commerceAccountHelper.getCurrentAccountEntry(
+				groupId, httpServletRequest);
```

Comment:
This can't be right. We need the channelGroupId and not the siteGroupId.
So you must call something like
commerceChannelLocalService.getCommerceChannelGroupIdBySiteGroupId(...) to convert it.

Reply:
ahh thanks for catching that!